src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -337,8 +337,24 @@ private void SetupAnalyzersForNewProject(string projectFullPath, IAnalysisContex
         public void ProcessEvaluationFinishedEventArgs(
             IAnalysisContext analysisContext,
             ProjectEvaluationFinishedEventArgs evaluationFinishedEventArgs)
-            => _buildEventsProcessor
-                .ProcessEvaluationFinishedEventArgs(analysisContext, evaluationFinishedEventArgs);
+        {
+            Dictionary<string, string>? propertiesLookup = null;
+            // The FileClassifier is normally initialized by executing build requests.
+            // However, if we are running in a main node that has no execution nodes - we need to initialize it here (from events).
```

Comment:
Good catch here!  
Now, I am not sure how this will work with msbuild server node though? Will this run on msbuild server node that has no execution nodes (if this scenario is at all supported, because it is pointless, I do not remember this detail).

Reply:
So up till now we needed FileClassifier only as part of the build execution - so on nodes that can execute requests. In case of MSBuild Server - if the main node (the 'server') had in-proc worker node - it already had the classifier instantiated, if it didn't have in-proc node - then classifier was not needed.
So luckily - no problem nor no impact for MSBuild Server. But good question indeed.

Reply:
I am not worried whether msbuild server is covered with FileClassifier. I am worried about analyzers. FileClassifier is needed for main non-server node without in-proc. Should it be needed for server main node without in-proc and whether it is covered. The scenario is weird - there is no reason to do that, but I am not sure it is prohibited. Also, there is a question of whether we tried analyzers with server feature and whether it works.